src/app/icd/ICDMonitoringTable.h

```diff
@@ -16,38 +16,75 @@
  */
 #pragma once
 
+#include <crypto/CHIPCryptoPAL.h>
+#include <crypto/SessionKeystore.h>
 #include <lib/core/CHIPConfig.h>
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/core/DataModelTypes.h>
 #include <lib/support/CodeUtils.h>
 #include <lib/support/PersistentData.h>
 #include <stddef.h>
 
+namespace chip {
+namespace Crypto {
+using SymmetricKeystore = SessionKeystore;
+}
+} // namespace chip
+
 namespace chip {
 
 inline constexpr size_t kICDMonitoringBufferSize = 40;
 
 struct ICDMonitoringEntry : public PersistentData<kICDMonitoringBufferSize>
 {
-    static constexpr size_t kKeyMaxSize = 16;
 
     ICDMonitoringEntry(FabricIndex fabric = kUndefinedFabricIndex, NodeId nodeId = kUndefinedNodeId)
     {
         this->fabricIndex      = fabric;
         this->checkInNodeID    = nodeId;
         this->monitoredSubject = nodeId;
     }
+
+    ICDMonitoringEntry(Crypto::SymmetricKeystore * keyStore, FabricIndex fabric = kUndefinedFabricIndex,
+                       NodeId nodeId = kUndefinedNodeId)
+    {
+        this->fabricIndex       = fabric;
+        this->checkInNodeID     = nodeId;
+        this->monitoredSubject  = nodeId;
+        this->symmetricKeystore = keyStore;
+    }
+
     bool IsValid() { return this->checkInNodeID != kUndefinedNodeId && this->fabricIndex != kUndefinedFabricIndex; }
     CHIP_ERROR UpdateKey(StorageKeyName & key) override;
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
     CHIP_ERROR Deserialize(TLV::TLVReader & reader) override;
     void Clear() override;
+    CHIP_ERROR SetKey(ByteSpan keyData);
+    CHIP_ERROR DeleteKey(void);
 
-    chip::FabricIndex fabricIndex = static_cast<chip::FabricIndex>(0);
-    chip::NodeId checkInNodeID    = static_cast<chip::NodeId>(0);
-    uint64_t monitoredSubject     = static_cast<uint64_t>(0);
-    chip::ByteSpan key;
-    uint16_t index = 0;
+    /**
+     * @brief Implement the key verification needed by the ICDManagement Server.
+     *        Since we cannot retrieve the key from the AES128KeyHandle IF using
+     *        PSA CRYPTO, we must implement a way to deduct if the verification key
+     *        received is the same or at least works as the same way as the one stored.
```

Comment:
```suggestion
     *        Since for some key implementations we cannot retrieve the key from the
     *        AES128KeyHandle, we must implement a way to deduce whether the verification key
     *        received is the same or at least works as the same way as the one stored.
```